### DISCUSSION TITLE
2017 style csproj

### DISCUSSION THREAD

**Discussion by rfcdejong** [MarimerLLC/csla#1822] [Ideas]
Currently I'm very occupied by converting all our old style csproj files from msbuild xml to readable csproj which was intruduced in VS2017. This has an advantage that our projects can be multi targetted to both .NET 462 and .NET Core. While we have over 100 projects in our solution (madness) it is not that hard gladly.

I want to ask why not all csla projects aren't new style csproj? I see Csla.Data.EFCore.3.0 is. Or better, why isn't there one single project that is doing multi targets like the example below. I know they shared projects and foreach framework there is an csproj which reference the shared project. It would had been easier to make a csproj multi target? 
It works but it could had been easier

An example of one of our csproj

```
<Project Sdk="Microsoft.NET.Sdk">
	<PropertyGroup>
	  <TargetFrameworks>net462;netcoreapp3.1</TargetFrameworks>
	  <RootNamespace>Dias.Framework.Data</RootNamespace>
	</PropertyGroup>

	<ItemGroup Condition=" '$(TargetFramework)' == 'net462' ">
	  <ProjectReference Include="..\..\..\Lib\Csla.Net4.6\Csla.Net4.6.csproj" />
	  <ProjectReference Include="..\..\..\Lib\Csla.Data.EF6.Net4.6\Csla.Data.EF6.Net4.6.csproj" />
	</ItemGroup>
	<ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp3.1' ">
	  <ProjectReference Include="..\..\..\Lib\Csla.AspNetCore.NetCore3.1\Csla.AspNetCore.NetCore3.1.csproj" />
	  <ProjectReference Include="..\..\..\Lib\Csla.Data.EFCore.3.0\Csla.Data.EFCore.3.0.csproj" />
	</ItemGroup>	

	<ItemGroup Condition=" '$(TargetFramework)' == 'net462' ">
	  <PackageReference Include="EFSecondLevelCache" version="1.2.0.0" />
	  <PackageReference Include="EntityFramework" version="6.1.3" />
	</ItemGroup>
	<ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp3.1' ">
	  <PackageReference Include="Microsoft.EntityFrameworkCore" Version="3.1.8" />
	</ItemGroup>

         ....
</Project>
```

**Comment by rfcdejong** [comment]
One valid reason I can think of for multiple projects is that they allow us to use EF6 inside .NET Core (version 6.4.4)
But this discussion is more about why not all the older projects could be converted to the new style.

**Reply by rockfordlhotka** [reply]
CSLA currently still supports .NET 4.0, and I think that's the only blocker (besides time and effort).

CSLA 6 will drop support for everything earlier than 4.7.2, at least that is the plan. So I think at that point the core Csla project could be reworked.

I don't know about the .NET Framework Windows Forms, EF, WPF, UWP, etc. projects - I'm not sure they can work with the new project file?

**Reply by rfcdejong** [reply]
This works for WPF, and I think it could work for Windows Forms, but I don't know about UWP.

```
<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
  <PropertyGroup>
    <TargetFrameworks>net462</TargetFrameworks>
    <UseWpf>true</UseWpf>
    <UseWindowsForms>true</UseWindowsForms>
  </PropertyGroup>
</Project>
```
